content/docs/pipelines/sdk/static-type-checking.md

```diff
@@ -0,0 +1,144 @@
+*'component a'* expects an input with Integer type and emits three outputs with type GCSPath, customized_type and GCRPath. 
+Among these tyeps, Integer, GCSPath, and GCRPath are core types that are predefined in the SDK while customized_type is a user-defined
+type.  
+```yaml
+name: component a
+description: component desc
+inputs:
+  - {name: field_l, type: Integer}
+outputs:
+  - {name: field_m, type: {GCSPath: {openapi_schema_validator: {type: string, pattern: "^gs://.*$" } }}}
+  - {name: field_n, type: customized_type}
+  - {name: field_o, type: GCRPath} 
+implementation:
+  container:
+    image: gcr.io/ml-pipeline/component-a
+    command: [python3, /pipelines/component/src/train.py]
+    args: [
+      --field-l, {inputValue: field_l},
+    ]
+    fileOutputs: 
+      field_m: /schema.txt
+      field_n: /feature.txt
+      field_o: /output.txt
+```
+Similarly, when authors write a component with the decorator, types are annotated in the function signature, as shown below.
+
+```python
+from kfp.dsl import component
+from kfp.dsl.types import Integer, GCRPath
+@component
+def task_factory_a(field_l: Integer()) -> {'field_m': {'GCSPath': {'openapi_schema_validator': '{"type": "string", "pattern": "^gs://.*$"}'}}, 
+                                           'field_n': 'customized_type',
+                                           'field_o': GCRPath()
+                                          }:
+    return ContainerOp(
+        name = 'operator a',
+        image = 'gcr.io/ml-pipeline/component-a',
+        command = [
+          'python3', '/pipelines/component/src/train.py'
+        ],
+        arguments = [
+            '--field-l', field_l,
+        ],
+        file_outputs = {
+            'field_m': '/schema.txt',
+            'field_n': '/feature.txt',
+            'field_o': '/output.txt'
+        }
+    )
+```
+Pipelines can also be annotated with types and being checked against the I/O types of the components. For example,
+```python
+@component
+def task_factory_a(field_m: {'GCSPath': {'openapi_schema_validator': '{"type": "string", "pattern": "^gs://.*$"}'}}, field_o: 'Integer'):
+    return ContainerOp(
+        name = 'operator a',
+        image = 'gcr.io/ml-pipeline/component-a',
+        arguments = [
+            '--field-l', field_m,
+            '--field-o', field_o,
+        ],
+    )
+
+# Pipeline input types are also checked against the component I/O types.
+@dsl.pipeline(name='type_check',
+    description='')
+def pipeline(a: {'GCSPath': {'openapi_schema_validator': '{"type": "string", "pattern": "^gs://.*$"}'}}='good', b: Integer()=12):
+    task_factory_a(field_m=a, field_o=b)
+
+try:
+    compiler.Compiler().compile(pipeline, 'pipeline.tar.gz', type_check=True)
+except InconsistentTypeException as e:
+    print(e)
+```
+
+## How does the type checking work?
+The basic checking criteria is the equality checking. In other words, type checking passes only when the type name strings are equal
```

Comment:
s/criteria/criterion

(criterion is the singular of criteria)

Reply:
Fixed.